cypress/integration/device_page_test.js

```diff
@@ -0,0 +1,36 @@
+describe('Device page tests', () => {
+  context('no access before login', () => {
+    it('redirects to home', () => {
+      cy.visit('/devices/PO6RuqIXQuysOCAJrcedqA');
+      cy.location('pathname').should('eq', '/login');
+    });
+  });
+
+  context('authenticated', () => {
+    beforeEach(() => {
+      cy.login();
+      cy.fixture('device').as('device');
+      cy.server();
+      cy.route('GET', '/appengine/v1/*/devices/*', '@device');
+      cy.route('GET', '/sockjs-node/**').as('websocket');
+    });
+
+    it('successfully loads', () => {
+      cy.visit('/devices/PO6RuqIXQuysOCAJrcedqA', {
+        onBeforeLoad: (window) => {
+          window.addEventListener('unhandledrejection', (event) => {
+            throw new Error(event.reason);
+          });
+        },
+      });
+      cy.location('pathname').should('eq', '/devices/PO6RuqIXQuysOCAJrcedqA');
+      cy.get('h2').contains('Device');
+      cy.wait(500);
+    });
+
+    it('requests a websocket connection', () => {
+      cy.visit('/devices/PO6RuqIXQuysOCAJrcedqA');
+      cy.wait('@websocket');
```

Comment:
this requires no interaction with the page, you can merge it in the previous test (saves loading times),
also see the previous message